lib/penthouse/runners/base_runner.rb

```diff
@@ -16,20 +16,32 @@ class BaseRunner
       # @return [void]
       # @raise [Penthouse::TenantNotFound] if the tenant cannot be switched to
       def self.call(tenant_identifier, &block)
-        load_tenant(tenant_identifier).call do |tenant|
+        previous_tenant_indentifier = call_stack.last || 'public'
+        call_stack.push(tenant_identifier)
+
+        load_tenant(tenant_identifier, previous_tenant_indentifier).call do |tenant|
           Penthouse.with_tenant(tenant.identifier) do
             block.yield(tenant)
           end
         end
+        
+        call_stack.pop
```

Comment:
This may screw up return values, is it worth using an `ensure` block for the `pop`?